cordovaPlatforms in package.json may have a duplicated platform entry

tq-jappy opened the pull request:
cordovaPlatforms in package.json may have a duplicated platform entry such as:

```json
  "cordovaPlatforms": [
    {
      "platform": "ios",
      "version": "3.7.0",
      "locator": "ios@3.7.0"
    },
    {
      "platform": "ios",
      "version": "3.8.0",
      "locator": "ios@3.8.0"
    }
  ],
```

## Reproduce

1. install platform with version ( ``ionic platform add ios@3.7.0`` )
1. remove platform directory **manually** ( ``rm -rf platforms/ios`` )
1. install platform with version again ( ``ionic platform add ios@3.8.0`` )

Then, platform is collectly installed, but package.json is not updated but added.

jbavari commented:
Thanks for the submission! Could I talk you into writing a unit test to cover this?

tq-jappy commented:
Okay, I added a unit test.

Regarding this subject, I found another issue in an existing unit test.
It seems to me that [the following assertion](https://github.com/driftyco/ionic-app-lib/blob/master/spec/state.spec.js#L99) is meaningless, because this will always be passed.

```
expect(defaultPackageJson).toBe(defaultPackageJson);
```

I also fixed it in this pull request.

jbavari commented:
Thank you, so much! You rock!

Ah, my bad - such an oversight. I appreciate your sharp eyes.
